File: sdk-kotlin/src/main/kotlin/org/sinou/pydia/sdk/utils/MemoryStore.java
package org.sinou.pydia.sdk.utils;

import java.util.HashMap;
import java.util.Map;

import org.sinou.pydia.sdk.api.Store;

public class MemoryStore<T> implements Store<T> {

    private final Map<String, T> objects = new HashMap<>();

    @Override
    public void put(String id, T obj) {
        objects.put(id, obj);
    }

    @Override
    public T get(String id) {
        return objects.get(id);
    }

    @Override
    public Map<String, T> getAll() {
        return objects;
    }

    @Override
    public void remove(String id) {
        objects.remove(id);
    }

    @Override
    public void clear() {
        objects.clear();
    }
}

File: sdk-kotlin/src/main/kotlin/org/sinou/pydia/sdk/utils/MemoryStore.kt
package org.sinou.pydia.sdk.utils

import org.sinou.pydia.sdk.api.Store

class MemoryStore<T> : Store<T> {

    private val objects: MutableMap<String, T> = HashMap()
    override fun put(id: String, obj: T) {
        objects[id] = obj
    }

    override fun get(id: String): T? {
        return objects[id]
    }

    override fun getAll(): Map<String, T> {
        return objects
    }

    override fun remove(id: String) {
        objects.remove(id)
    }

    override fun clear() {
        objects.clear()
    }
}